### PR TITLE
[Mongodb Atlas] Organization Log Datastream

### PR DESCRIPTION
**Pull request by milan-elastic**
Create the package for Mongodb Atlas Integration and Provide the support of **Organization logs**.

# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

Along with the organization PR, resolved the following issue for process, mongod_audit and mongod_database as well.
- https://github.com/elastic/obs-infraobs-team/issues/1372

### All changes

- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/main/CONTRIBUTING.md)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists, [useful guidelines](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md) to follow
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: 8.12.0
- [ ] TSDB enablement for the data stream

### Data dataset changes

_This entry is currently only recommended. It will be mandatory once we provide better support for it._

- [x] Sample event (`sample_event.json`) exists

### Related Issue
- Relates https://github.com/elastic/obs-infraobs-team/issues/1375

### Known Issues:
Currently Following challenges are still open to solve for this PR:
- https://github.com/elastic/obs-infraobs-team/issues/1350
- https://github.com/elastic/obs-infraobs-team/issues/1351
- https://github.com/elastic/obs-infraobs-team/issues/1372

### CODE REVIEW

File: packages/mongodb_atlas/data_stream/organization/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        description: The request tracer logs requests and responses to the agent's local file-system for debugging configurations. Enabling this request tracing compromises security and should only be used for debugging. See [documentation](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-cel.html#_resource_tracer_filename) for details.
+      - name: tags
+        type: text
+        title: Tags
+        multi: true
+        required: true
+        show_user: false
+        default:
+          - mongodb_atlas-organization
+      - name: preserve_original_event
```
Comment (niraj-elastic) [inline packages/mongodb_atlas/data_stream/organization/manifest.yml:47]: We have not handled `preserve orignal` in pipeline, so we can either remove it from here, or add it in pipeline.

**Review by niraj-elastic — APPROVED** [review]
LGTM

File: packages/mongodb_atlas/data_stream/organization/sample_event.json [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    "mongodb_atlas": {
+        "organization": {
+            "access_list_entry": "sample.user@example.com",
+            "additional_info": {
+                "_t": "RESOURCE_AUDIT",
+                "cid": "663087fcc4818d301a53af06",
+                "cre": "2024-04-30T06:17:35Z",
+                "description": "Tag(s) were added or modified on project",
+                "gn": "test_project_org",
+                "orgName": "Integrations - 2023-05-25",
```
Comment (muthu-mps) [inline packages/mongodb_atlas/data_stream/organization/sample_event.json:80]: Can we convert the JSON keys to snake_case?

**Reference:** https://github.com/elastic/integrations/blob/8fad2d63b9a38c89f64e0edc8a0f4af367fc692f/packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml#L15

File: packages/mongodb_atlas/data_stream/organization/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      fields:
+        - name: id
+          type: keyword
+          description: ID of the cluster to which this event applies.
+        - name: name
+          type: keyword
+          description: Name of the cluster to which this event applies.
+    - name: collection.name
+      type: keyword
+      description: Name of the collection on which the event occurred.
```
Comment (shmsr) [inline packages/mongodb_atlas/data_stream/organization/fields/fields.yml:33]:
```suggestion
      description: Name of the collection where the event occurred.
```

File: packages/mongodb_atlas/data_stream/organization/fields/fields.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      fields:
+        - name: number
+          type: double
+          description: The value of the metric at the time of the event.
+        - name: units
+          type: keyword
+          description: Relevant units for the value.
+    - name: database.name
+      type: keyword
+      description: Name of the database on which the event occurred.
```
Comment (shmsr) [inline packages/mongodb_atlas/data_stream/organization/fields/fields.yml:45]:
```suggestion
      description: Name of the database where the event occurred.
```

File: packages/mongodb_atlas/data_stream/organization/fields/fields.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    - name: event_type.name
+      type: keyword
+      description: Indicates name of the event.
```
Comment (shmsr) [inline packages/mongodb_atlas/data_stream/organization/fields/fields.yml:48]:
```suggestion
      description: Name of the event type.

```

File: packages/mongodb_atlas/data_stream/organization/fields/fields.yml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    - name: host
+      type: group
+      fields:
+        - name: id
+          type: keyword
+          description: ID of the host on which this event occurred.
```
Comment (shmsr) [inline packages/mongodb_atlas/data_stream/organization/fields/fields.yml:54]:
```suggestion
          description: ID of the host where this event occurred.
```

File: packages/mongodb_atlas/data_stream/organization/fields/fields.yml (hunk 5 above)
Comment (shmsr) [reply packages/mongodb_atlas/data_stream/organization/fields/fields.yml:48]: Probably avoid the word "Indicates" as we are not using this for other fields. Only a few are using "Indicates".

**Review by shmsr — CHANGES_REQUESTED** [review]

File: packages/mongodb_atlas/_dev/build/docs/README.md [hunk 7]
```diff
@@ -38,22 +37,24 @@ You can store and search your data using Elasticsearch and visualize and manage
...
-2. From the Atlas UI, go to Project Settings > Access Manager > API Keys and then click on Invite To Project to add the API key created above.
-3. Add specific role to API keys, under Project Settings > Access Manager > API Keys. This step is important to make sure that these API keys have the right permissions to access the data without running into any issues. The specific role for each datastream is defined under data stream reference section.
+1. Generate programmatic API keys with `Organization Owner` permission by following the instructions in the Atlas [documentation](https://www.mongodb.com/docs/atlas/configure-api-access/#grant-programmatic-access-to-an-organization). Then, copy the public and private keys which function as a username and API key respectively.
+2. From the Atlas UI with `project owner` permission, go to Project Settings > Access Manager > API Keys and then click on Invite To Project to add the API key created above, as described in this Atlas [document](https://www.mongodb.com/docs/atlas/configure-api-access/#invite-an-organization-api-key-to-a-project).
```
Comment (alaudazzi) [inline packages/mongodb_atlas/_dev/build/docs/README.md:46]: UI text and components are bold
```suggestion
2. From the Atlas UI with `project owner` permission, go to *Project Settings* > *Access Manager* > *API Keys* and then click *Invite To Project* to add the API key created above, as described in the Atlas [documentation](https://www.mongodb.com/docs/atlas/configure-api-access/#invite-an-organization-api-key-to-a-project).
```

**Review by alaudazzi — APPROVED** [review]
Left a few minor editing suggestions, otherwise LGTM.

**Comment by muthu-mps** [comment]
@milan-elastic - The header values for the table visualisation can be updated to make it more readable. Check whether we could rename those values.

Screenshot 2024-05-09 at 2 14 40 PM

**Review by shmsr — APPROVED** [review]

**Comment by milan-elastic** [comment]
> @milan-elastic - The header values for the table visualisation can be updated to make it more readable. Check whether we could rename those values.
> 
> Screenshot 2024-05-09 at 2 14 40 PM

@muthu-mps It is saved search, hence we can not change the header of it!

**Review by muthu-mps — APPROVED** [review]
@ishleenk17 - Please have a second view on dashboards. We can think of adding a overview dashboard later.
Change looks good to me!

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**